client/src/components/Profile.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { Component } from 'react';
+import { Row, Col, Input, Button } from 'react-materialize';
+import { server_url } from '../config.json';
+
+
+export default class Profile extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            edit: false,
+            user: []
+        };
+
+        this.Avatar_image = this.Avatar_image.bind(this);
+        this.UpdateClick = this.UpdateClick.bind(this);
+        this.onChangeHandler = this.onChangeHandler.bind(this);
+        this.handleClick_Main = this.handleClick_Main.bind(this);
+        this.handleClick_Edit = this.handleClick_Edit.bind(this);
+    }
+
+    componentDidMount() {
+        fetch(server_url + '/profile/' + window.location.href.split("/").pop())
+        .then(response => response.json())
+        .then(json => {
+            this.setState({ user: json });
+            console.log(json)
```

Comment:
Remove console.log...

Reply:
About NavProfile's routeChange:

I just tested your solution while being In the profile itself and it returns this error:
Warning: Hash history cannot PUSH the same path; a new entry will not be added to the history stack